service.py

```diff
@@ -174,11 +187,16 @@ def find_tv_show_season(content, tvshow, season):
 def append_subtitle(item):
     title = item['filename']
     if 'comment' in item and item['comment'] != '':
-        title = "%s [COLOR gray][I](%s)[/I][/COLOR]" % (title, item['comment'])
-    listitem = xbmcgui.ListItem(label=item['lang']['name'],
-                                label2=title,
-                                iconImage=item['rating'],
-                                thumbnailImage=item['lang']['2let'])
+        title = "%s[CR][COLOR silver][I](%s)[/I][/COLOR]" % (title, item['comment'])
+    if sys.version_info.major == 3:
+        listitem = xbmcgui.ListItem(label=item['lang']['name'],
```

Comment:
Can you explain why these things need to be separated between Python 3 and 2? Is xbmcgui API different between these Python versions?

Reply:
This change was due to WARNING messages in the logs:
WARNING: Using iconImage in ListItem constructor results in NOP. Use setArt.
WARNING: Using thumbnailImage in ListItem constructor results in NOP. Use setArt.

I don't have a Kodi18 or Python2 version to test. The flag images and stars were not displayed unless the call was changed. I assume it works fine on your version. While checking for the API version may be more elegant (no idea how to do that though), the main difference between Kodi 18 and 19 is the python version, so this should be fine either way?

Reply:
Yes, these things worked for me on Kodi 18. However, I didn't know that Kodi bundles Python with itself - is that true? I'm running Kodi 18 on my Raspberry Pi and see that there's Python 2 and 3 installed - I guess it uses Python 2 then, but how do I know? What would happen if I would remove Python 2 and only have Python 3 set up there? I can imagine that Kodi would try to use that (and fail spectacularly?)?

Reply:
Kodi bundles Python and includes all the standard python modules, any additional modules will have to be kodi addons rather than system python addons. Kodi started bundling Python 3 in the Kodi 19 nightlies a couple of months ago. Kodi 18 and old Kodi 19 nightlies ship with Python 2.
Removing the system python version should not make much of a difference apart from breaking your system itself as there are too many system dependencies on python 2. I believe there are plans to remove python 2 from Ubuntu next year.